Source/console-checkers/Source/CheckersNotationView.cpp

```diff
@@ -0,0 +1,196 @@
+//---------------------------------------------------------------
+//
+// CheckersNotationView.cpp
+//
+
+#include "CheckersNotationView.h"
+
+#include "GameSettings.h"
+#include "UITextStrings.h"
+#include "Utility.h"
+
+#include <spdlog/spdlog.h>
+#include <spdlog/fmt/bundled/format.h>
+
+#include <sstream>
+
+namespace Checkers {
+
+//===============================================================
+
+// ANSI escape code info can be found here, explains what the values 48, 38, 5, etc are:
+// https://en.wikipedia.org/wiki/ANSI_escape_code
+
+// Environment colors.
+static constexpr std::string_view s_pieceBackgroundColor = "\033[48;5;233m";
+static constexpr std::string_view s_hintBackgroundColor = "\033[48;5;228m";
+static constexpr std::string_view s_resetColor = "\033[0m";
+
+// Piece colors
+static constexpr std::string_view s_redPawnColor = "\033[38;5;196m";
+static constexpr std::string_view s_redKingColor = "\033[38;5;166m";
+static constexpr std::string_view s_blackPawnColor = "\033[38;5;33m";
+static constexpr std::string_view s_blackKingColor = "\033[38;5;117m";
+static constexpr std::string_view s_emptySquareColor = "\033[38;5;102m";
+
+static constexpr std::string_view s_redPawnColorHint = "\033[38;5;160m";
+static constexpr std::string_view s_redKingColorHint = "\033[38;5;202m";
+static constexpr std::string_view s_blackPawnColorHint = "\033[38;5;24m";
+static constexpr std::string_view s_blackKingColorHint = "\033[38;5;74m";
+static constexpr std::string_view s_emptySquareColorHint = "\033[38;5;239m";
+
+// You only see this if there's an error with finding the piece color.
+static constexpr std::string_view s_errorColor = "\033[38;5;206m";
+
+// In Checkers notation, the first numbered position is 1.
+static constexpr int32_t s_positionStart = 1;
+
+static std::string_view GetColorForPiece(const Piece& piece, bool shouldShowHint)
+{
+	static const std::unordered_map<Piece, std::string_view, PieceHash> s_colorMap =
+	{
+		{ { PieceType::Empty, Identity::Neutral }, s_emptySquareColor },
+		{ { PieceType::Pawn, Identity::Red }, s_redPawnColor },
+		{ { PieceType::King, Identity::Red}, s_redKingColor },
+		{ { PieceType::Pawn, Identity::Black }, s_blackPawnColor },
+		{ { PieceType::King, Identity::Black }, s_blackKingColor },
+	};
+
+	static const std::unordered_map<Piece, std::string_view, PieceHash> s_colorHighlightMap =
+	{
+		{ { PieceType::Empty, Identity::Neutral }, s_emptySquareColorHint },
+		{ { PieceType::Pawn, Identity::Red }, s_redPawnColorHint },
+		{ { PieceType::King, Identity::Red}, s_redKingColorHint },
+		{ { PieceType::Pawn, Identity::Black }, s_blackPawnColorHint },
+		{ { PieceType::King, Identity::Black }, s_blackKingColorHint },
+	};
+
+	const auto& relevantColorMap = shouldShowHint ? s_colorHighlightMap : s_colorMap;
+	const auto it = relevantColorMap.find(piece);
+	if (it != relevantColorMap.end())
+	{
+		return it->second;
+	}
+
+	return s_errorColor;
```

Comment:
Yeah this is a common thing in game design that I've noticed. The `s_errorColor` in this case is set to a bright pink which is typical for indicating an "unset" or "default" color that isn't supposed to ship. That way when something is not setup properly its really obvious.

Reply:
Ya after I made this comment I thought about "Shane Cubes" for WoW. Games are SO SO SO keen on making sure it runs no matter what that it is completely valid to run without an asset. However in this context you KNOW all pieces and colors at compile time. So I would turn this around and say "Make selecting the wrong color a compile error instead of a runtime error".